app/services/catalog/service_plans.rb

```diff
@@ -9,23 +11,41 @@ def process
       ref = PortfolioItem.find(@portfolio_item_id).service_offering_ref
       TopologicalInventory.call do |api_instance|
         result = api_instance.list_service_offering_service_plans(ref)
-        @items = filter_result(result.data)
+        @items = filter_result(result.data, ref)
       end
       self
     rescue StandardError => e
       Rails.logger.error("Service Plans #{e.message}")
       raise
     end
 
-    def filter_result(result)
-      result.collect do |obj|
-        {
-          'name'               => obj.name,
-          'description'        => obj.description,
-          'id'                 => obj.id,
-          'create_json_schema' => obj.create_json_schema
-        }
+    private
+
+    def filter_result(result, ref)
+      if result.empty?
+        no_service_plan(ref)
+      else
+        result.collect do |obj|
+          {
+            'name'               => obj.name,
+            'description'        => obj.description,
+            'id'                 => obj.id,
+            'create_json_schema' => obj.create_json_schema
+          }
+        end
       end
     end
+
+    def no_service_plan(ref)
```

Comment:
These two custom hashes really belong in a view and not in a service.  I know we don't currently do JSON building from the view, but in this case rendering a json payload from something like app/services/views segregates these custom entries away from the logic used to build them. 

We already have jbuilder which could work nice for this: https://github.com/rails/jbuilder